File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/mapping/RequestMapper.java
package com.ygo74.agentruntime.domains.mapping;

import com.ygo74.agentruntime.domains.contracts.ExchangeModels.StandardExchangeRequest;
import com.ygo74.agentruntime.domains.endpoints.EndpointAdapters;

import java.util.Map;

public final class RequestMapper {
    private RequestMapper() {}

    public static StandardExchangeRequest toExchange(
            String endpointType,
            String requestId,
            String routeKey,
            Object input,
            boolean stream,
            Map<String, Object> metadata) {
        return EndpointAdapters.normalize(endpointType, requestId, routeKey, input, stream, metadata);
    }
}

File: packages/dotnet/Ygo74.AgentRuntime/Domains/Mapping/RequestMapper.cs
using Ygo74.AgentRuntime.Domains.Contracts;
using Ygo74.AgentRuntime.Domains.Endpoints;

namespace Ygo74.AgentRuntime.Domains.Mapping;

public static class RequestMapper
{
    public static StandardExchangeRequest ToExchange(string endpointType, string requestId, string routeKey, object input, bool stream = false, IDictionary<string, object?>? metadata = null)
        => EndpointAdapters.Normalize(endpointType, requestId, routeKey, input, stream, metadata);
}
